Source/Diagnostics/Diagnostics.H

```diff
@@ -18,99 +18,125 @@ class Diagnostics
+     * \param[in] lev level on which the output MultiFab is defined
+     */
+    virtual void InitializeFieldBufferData (int i_buffer, int lev) = 0;
+    /** Initialize member variables and arrays specific to the diagnostics in the
+     *  derived classes.(FullDiagnostics, BTDiagnostics)
      */
-    void AddRZModesToOutputNames (const std::string& field, int ncomp);
-    /** Append m_all_field_functors[lev] with all modes of all components of E B and j
-     * and add name to varnames.
-     * \param[in] lev level on which source multifabs are defined.
+    virtual void DerivedInitData () {};
+    /** This function initialized particle buffers (not implemented in diagnostics, yet) */
+    virtual void InitializeParticleBuffer () = 0;
+    /** Prepare data (either fill-boundary or cell-centered data for
+        back-transform diagnostics) to be processed for diagnostics.
      */
-    void AddRZModesToDiags (int lev);
+    virtual void PrepareFieldDataForOutput () {};
     /** Name of diagnostics: runtime parameter given in the input file. */
     std::string m_diag_name;
     /** Prefix for output directories */
     std::string m_file_prefix;
     /** Index of diagnostics in MultiDiagnostics::alldiags */
     int m_diag_index;
-    /** Determines timesteps at which diagnostics are written to file */
-    IntervalsParser m_intervals;
     /** Names of  each component requested by the user.
      * in cylindrical geometry, this list is appended with
      * automatically-constructed names for all modes of all fields */
     amrex::Vector< std::string > m_varnames;
-    /** Vector of (pointers to) functors to compute output fields, per level,
-     * per component. This allows for simple operations (averaging to
-     * cell-center for standard EB fields) as well as more involved operations
-     * (back-transformed diagnostics, filtering, reconstructing cartesian
-     * fields in cylindrical). */
-    amrex::Vector< amrex::Vector <std::unique_ptr<ComputeDiagFunctor const> > > m_all_field_functors;
-    /** output multifab, where all fields are cell-centered and stacked */
-    amrex::Vector< amrex::MultiFab > m_mf_output;
-    int nlev; /**< number of levels to output */
-    int nmax_lev; /**< max_level to allocate output multifab and vector of field functors. */
-    /** This class is responsible for flushing the data to file */
-    FlushFormat* m_flush_format;
-    /** Whether to plot raw (i.e., NOT cell-centered) fields */
-    bool m_plot_raw_fields = false;
-    /** Whether to plot guard cells of raw fields */
-    bool m_plot_raw_fields_guards = false;
-    /** Whether to plot charge density rho in raw fields */
-    bool m_plot_raw_rho = false;
-    /** Whether to plot F (charge conservation error) in raw fields */
-    bool m_plot_raw_F = false;
-    /** Whether to dump the RZ modes */
-    bool m_dump_rz_modes = false;
+    /** format for output files, "plotfile" or "openpmd" */
```

Comment:
Is this more yet? I.e. `ascent` and `sensei`, too?
```suggestion
    /** format for output files, "plotfile" or "openpmd" */
```

Reply:
Thank you for noticing this. I will add ascent and sensei too. Currently BTD does not support plotfile/openpmd/ascent/sensei and so I am adding Asserts for these formats in BTDiagnostics.cpp